cdsodatacli/query.py

```diff
@@ -42,22 +50,23 @@ def fetch_data(gdf=None, date=None, dtime=None, timedelta_slice=None, start_date
                                  dtime=dtime,
                                  timedelta_slice=timedelta_slice)
         # geopd_norm.sort_index(ascending=False)
-        print(gdf_norm)
-        print(f"Length of input:{len(gdf_norm)}")
+        logging.debug(f"Length of input:{len(gdf_norm)}")
         urls = urls_creat(gdf=gdf_norm, top=top)
 
-    collected_data = fetch_data_from_urls(urls=urls)
-    # Remove duplicates
-    data_dedup = remove_duplicates(safes_ori=collected_data)
+        collected_data = fetch_data_from_urls(urls=urls)
+        if collected_data.empty:
+            return logging.info("No data found.")
 
-    # Convert all Multipolygon to Polygon and add geometry as new column
-    full_data = multy_to_poly(collected_data=data_dedup)
+        # Remove duplicates
+        data_dedup = remove_duplicates(safes=collected_data)
 
-    if min_sea_percent is not None:
-        full_data = sea_percent(collected_data=full_data, min_sea_percent=min_sea_percent)
+        # Convert all Multipolygon to Polygon and add geometry as new column
+        full_data = multi_to_poly(collected_data=data_dedup)
+        if min_sea_percent is not None:
+            full_data = sea_percent(collected_data=full_data, min_sea_percent=min_sea_percent)
 
-    if fig is not None:
-        fig(collected_data=full_data)
+        if figure:
```

Comment:
I am not fan with function to do figures in this library. I propose to remove it.

Reply:
So you mean we don't need to show the figure?